src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -220,10 +223,15 @@ private TensorFlowTransform(IHostEnvironment env, byte[] modelBytes, string[] in
                 var tfInput = new TFOutput(Graph[Inputs[i]]);
                 TFInputTypes[i] = tfInput.OutputType;
                 TFInputShapes[i] = Graph.GetTensorShape(tfInput);
-                var newShape = new long[TFInputShapes[i].NumDimensions];
-                for (int j = 0; j < TFInputShapes[i].NumDimensions; j++)
-                    newShape[j] = TFInputShapes[i][j] == -1 ? BatchSize : TFInputShapes[i][j];
-                TFInputShapes[i] = new TFShape(newShape);
+                if (TFInputShapes[i].NumDimensions != -1)
+                {
+                    var newShape = new long[TFInputShapes[i].NumDimensions];
+                    newShape[0] = TFInputShapes[i][0] == -1 ? BatchSize : TFInputShapes[i][0];
```

Comment:
>newShape[0] = TFInputShapes[i][0] == -1 ? BatchSize : TFInputShapes[i][0]; [](start = 20, length = 74)

so we will have special handling only for the 1st dimension, and not for the other dimensions -- is that the intent ?

(looks like we should have been doing this previously too, instead of doing special handling for all the columns)

Reply:
yeap. good catch!

---
In reply to: [216477499](https://github.com/dotnet/machinelearning/pull/857#discussion_r216477499) [](ancestors = 216477499,216461730)